CONTRIBUTING.md

```diff
@@ -194,3 +186,86 @@ repository, we suggest doing some cleanup of these branches.
   ```console
   $ git push {username} :<branchname>
   ```
+
+## Commit Signoffs
+
+In order for us to accept your patches, you must provide a signoff in all
+commits; this is done by using the `-s` or `--signoff` option when calling `git
+commit`. The signoff is used to certify that you have rights to submit your
+patch under the project's license, and that you agree to the [Developer
+Certificate of Origin](https://developercertificate.org).
+
+### Automating signoffs
+
+You can automate adding your signoff by using a commit template that includes
+the line:
+
+```text
+Signed-off-by: Your Name <your.email@example.com>
+```
+
+Put that line into a file, and then run the command:
+
+```console
+git config commit.template {path to file}
+```
+
+If you want to use this template everywhere, pass the `--global` option to
+`git config`.
+
+### Adding signoffs when committing from the GitHub website
+
+You can add a signoff manually when committing from the GitHub website by adding
+the line `Signed-off-by: Your Name <your.email@example.org>` by itself in the
+commit message.
+
+### Adding signoffs after-the-fact
+
+If you have forgotten signoff your commits, you can do so as follows:
+
+- Determine the number of commits you have made.
+- Run `git rebase -i HEAD~{number you discovered}`
```

Comment:
## Variant 1:
`git rebase --signoff HEAD~{number you discovered}` is an easier option if all rebasing commits belong to the author.
## Variant 2:
`git rebase -i HEAD~{number you discovered} -x "git commit --amend --signoff --no-edit"` would present rebase todo with exec after each commit already in place:
```
pick fc1190f3 Rewriting as Laminas Project package
exec git commit --amend --signoff --no-edit
pick bb169095 Add PHP 7.4 to travis build matrix
exec git commit --amend --signoff --no-edit
pick be70e505 Add PHPUnit 7 for php 7.4
exec git commit --amend --signoff --no-edit   
```
Delete exec lines after commits that should not have signoff.
I did not check but I believe signoff line would be added only if same line is not present already.

---

Instructions on how to recover from rebase mistake might be helpful:

Run `git log -1` before rebase and make note of the commit hash
If something goes wrong during rebase, use `git rebase --abort` to return to original state.
If rebase already completed, use `git reset --hard {commit hash from before rebase}`

Reply:
I've incorporated this feedback - thanks for the pointers on `git rebase --signoff` and the `-x` flag when doing an interactive rebase!

I've NOT done any information on recovering from a rebase mistake other than mentioning `git rebase --abort`. I think it's likely better users get that information from the git manual.